### PR TITLE
Universal Projector Remote

### PR DESCRIPTION
**Pull request by itsweekly**
# What's new

- Added a universal project remote

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by xMasterX** [comment]
Where you got that file `assets/resources/infrared/assets/projectors.ir` ?
Also this is funny how "your" code still has my comment "//TODO Improve Projectors universal remote"
Is it fair to do copypasta like this and don't give a single credit to actual authors? For example `projectors.ir` is made by @amec0e 
and has totally different structure from .ir databases that exists in this repo

File: applications/main/infrared/infrared_cli.c [hunk 1]
```diff
@@ -86,7 +86,7 @@ static void infrared_cli_print_usage(void) {
     printf("\tir universal <remote_name> <signal_name>\r\n");
     printf("\tir universal list <remote_name>\r\n");
     // TODO: Do not hardcode universal remote names
-    printf("\tAvailable universal remotes: tv audio ac\r\n");
+    printf("\tAvailable universal remotes: tv audio ac projectors\r\n");
```
Comment (skotopes) [inline applications/main/infrared/infrared_cli.c]: We don't use plural form in file naming.

File: applications/main/infrared/scenes/infrared_scene_universal.c [hunk 2]
```diff
@@ -21,18 +22,35 @@ void infrared_scene_universal_on_enter(void* context) {
...
         context);
+
+    submenu_add_item(
+        submenu,
+        "Projectors",
+        SubmenuIndexUniversalProjector,
+        infrared_scene_universal_submenu_callback,
+        context);
+
+    submenu_add_item(
```
Comment (skotopes) [inline applications/main/infrared/scenes/infrared_scene_universal.c]: Fans are clearly shouldn't be part of this PR.

File: assets/resources/infrared/assets/projectors.ir [hunk 3]
```diff
@@ -0,0 +1,778 @@
+Filetype: IR library file
+Version: 1
+#
+# ON
+name: POWER
```
Comment (skotopes) [inline assets/resources/infrared/assets/projectors.ir]: Button naming doesn't match our standard schema

File: documentation/UniversalRemotes.md [hunk 4]
```diff
@@ -25,6 +25,28 @@ Make sure that every signal does what it's supposed to.
 
 If everything checks out, append these signals **to the end** of the [audio player universal remote file](/assets/resources/infrared/assets/audio.ir).
 
+## Projectors
```
Comment (skotopes) [inline documentation/UniversalRemotes.md:28]: Most of this instruction is irrelevant, just check previous section

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
@itsweekly you need to answer all review question and approve from @gsurkov before this PR will be merged.
Un-draft when ready.

**Comment by itsweekly** [comment]
> Where you got that file `assets/resources/infrared/assets/projectors.ir` ? Also this is funny how "your" code still has my comment "//TODO Improve Projectors universal remote" Is it fair to do copypasta like this and don't give a single credit to actual authors? For example `projectors.ir` is made by @amec0e and has totally different structure from .ir databases that exists in this repo

1. https://github.com/RandomDebugError/irdb and yes the .ir is @amec0e and yes I did forget 
2. Yes, your comment is still there because I copied one of the other universal_name.c files to get the structure then took my time to actually understand it
3. The structure of the .ir file Is very much the same except from the uppercase names (I have changed this)
@skotopes Didnt even realise I left fans in there, I did this on my own local copy which already had changed (fixed)

File: applications/main/infrared/infrared_cli.c (hunk 1 above)
Comment (itsweekly) [reply applications/main/infrared/infrared_cli.c]: Fixed

File: applications/main/infrared/scenes/infrared_scene_universal.c (hunk 2 above)
Comment (itsweekly) [reply applications/main/infrared/scenes/infrared_scene_universal.c]: Yep, missed out on it as I did this on my own copy which I had tampered on

File: assets/resources/infrared/assets/projectors.ir (hunk 3 above)
Comment (itsweekly) [reply assets/resources/infrared/assets/projectors.ir]: renamed

File: documentation/UniversalRemotes.md (hunk 4 above)
Comment (itsweekly) [reply documentation/UniversalRemotes.md:28]: Fixed

File: documentation/UniversalRemotes.md [hunk 5]
```diff
@@ -25,6 +25,10 @@ Make sure that every signal does what it's supposed to.
...
+
+To add your Projector assign it one of these names `POWER`, `MUTE`, `VOL+`, `VOL-`.
```
Comment (skotopes) [inline documentation/UniversalRemotes.md]: this one too

**Review by skotopes — CHANGES_REQUESTED** [review]

File: documentation/UniversalRemotes.md (hunk 5 above)
Comment (itsweekly) [reply documentation/UniversalRemotes.md]: > this one too

okay I fixed it

**Comment by amec0e** [comment]
@itsweekly Hi there, it seems that the new additions you added, the first two are duplicate entries which are already in the now outdated `projectors.ir` file. It seems your "Smart Board" addition I could not seem to locate on both @UberGuidoZ IRDB repo, nor Logicksworkshop IRDB repo (Which has the additions of the repo you mentioned under the `_Converted_` section. 

I was able to locate on Ubers Flipper-IRDB only one file that this code belongs to and that is a Vizio Soundbar located here: https://github.com/UberGuidoZ/Flipper-IRDB/blob/main/SoundBars/Vizio/Vizio_SB3651_E6.ir#L42

@skotopes and @gsurkov You are free to use my updated one located here: https://github.com/amec0e/Flipper/blob/main/Infrared/projectors.ir if you wish to do so, if you would like i can mass rename the buttons in the ir file to fit your naming scheme too. Currently it includes `VOL-, VOL+, MUTE, POWER`

**Comment by UberGuidoZ** [comment]
Bummer that this was just closed. I'd like to see these updates in Official but in a better capacity.

**Comment by itsweekly** [comment]
> @itsweekly Hi there, it seems that the new additions you added, the first two are duplicate entries which are already in the now outdated `projectors.ir` file. It seems your "Smart Board" addition I could not seem to locate on both @UberGuidoZ IRDB repo, nor Logicksworkshop IRDB repo (Which has the additions of the repo you mentioned under the `_Converted_` section.
> 
> I was able to locate on Ubers Flipper-IRDB only one file that this code belongs to and that is a Vizio Soundbar located here: https://github.com/UberGuidoZ/Flipper-IRDB/blob/main/SoundBars/Vizio/Vizio_SB3651_E6.ir#L42
> 
> @skotopes and @gsurkov You are free to use my updated one located here: https://github.com/amec0e/Flipper/blob/main/Infrared/projectors.ir if you wish to do so, if you would like i can mass rename the buttons in the ir file to fit your naming scheme too. Currently it includes `VOL-, VOL+, MUTE, POWER`

Woah, so from now I'm not free to use open databases and add to OFW? Public archives and even well maintained repos are just laying around there for no absolute reason? I'm not getting pressed or anything, just including you I don't think you created those repos just for the sake of getting that role of being a contributor. If you want to take over this request, go ahead, but don't do it in a cocky way

**Comment by itsweekly** [comment]
> Bummer that this was just closed. I'd like to see these updates in Official but in a better capacity.

For once thank you somebody does not shit on this request

**Comment by amec0e** [comment]
> Woah, so from now I'm not free to use open databases and add to OFW? Public archives and even well maintained repos are just laying around there for no absolute reason? I'm not getting pressed or anything, just including you I don't think you created those repos just for the sake of getting that role of being a contributor. If you want to take over this request, go ahead, but don't do it in a cocky way


Where did I say you couldn't use it? And where is the cocky manner in which I've supposed to have said this? I'm simply notifying you the additions you made are already in the file (duplicates) and the entry that is not is for a soundbar hence why it's not in the file. Which is now outdated as I updated this not long ago meaning you'll need to get a fresh copy...

I offered my assistance to both skotopes and gsurkov as the files 
original creator purely because if they did not want to use the one provided (they may want to add more buttons than what we currently have in the file) then I could generate them their own `projector.ir` list specific to the buttons they would like to include and if they did want to use the new one provided I could have renamed them all to fit their naming scheme. Was I not supposed to do this? Am I not allowed to?

As for the repo, your correct I did not make it nor was I taking credit for making it nor was it for any sort of status, I uploaded it to my fork as @UberGuidoZ is busy and might not have gotten around to merging the PR into his repo in time for me to post the comment.

**Comment by UberGuidoZ** [comment]
I agree @amec0e - it’s important to keep things clean as a lot of work has gone into maintaining it. (Mostly by you!)

I would still like to see these changes in OFW but trust there are conversations happening that likely aren’t here. ¯\\\_(ツ)_/¯

**Comment by skotopes** [comment]
> I agree @amec0e - it’s important to keep things clean as a lot of work has gone into maintaining it. (Mostly by you!)
> 
> I would still like to see these changes in OFW but trust there are conversations happening that likely aren’t here. ¯_(ツ)_/¯

We are ok to merge it.

**Comment by UberGuidoZ** [comment]
Awesome! But please do see the notes and changes from @amec0e on the updates.

**Comment by itsweekly** [comment]
> > I agree @amec0e - it’s important to keep things clean as a lot of work has gone into maintaining it. (Mostly by you!)
> > I would still like to see these changes in OFW but trust there are conversations happening that likely aren’t here. ¯_(ツ)_/¯
> 
> We are ok to merge it.

If this is still up, soon I will merge @amec0e IR with this, then add some of mine which I have collected, and just clear it up

**Comment by skotopes** [comment]
@itsweekly sure.

Please un-draft when ready.

**Comment by amec0e** [comment]
> If this is still up, soon I will merge @amec0e IR with this, then add some of mine which I have collected, and just clear it up

Cool.

File: applications/main/infrared/scenes/infrared_scene_universal_projector.c [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+#include "common/infrared_scene_universal_common.h"
+
+void infrared_scene_universal_projector_on_enter(void* context) {
+    infrared_scene_universal_common_on_enter(context);
+
+    Infrared* infrared = context;
+    ButtonPanel* button_panel = infrared->button_panel;
+    InfraredBruteForce* brute_force = infrared->brute_force;
+
+    infrared_brute_force_set_db_filename(brute_force, EXT_PATH("infrared/assets/projectors.ir"));
```
Comment (gsurkov) [inline applications/main/infrared/scenes/infrared_scene_universal_projector.c]: The file name should be `projector.ir`, otherwise the CLI won't work.

File: applications/main/infrared/scenes/infrared_scene_universal_projector.c [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        1,
+        36,
+        64,
+        &I_Vol_down_25x27,
+        &I_Vol_down_hvr_25x27,
+        infrared_scene_universal_common_item_callback,
+        context);
+    infrared_brute_force_add_record(brute_force, i++, "Vol_dn");
+
+    button_panel_add_label(button_panel, 10, 11, FontPrimary, "Projector");
```
Comment (gsurkov) [inline applications/main/infrared/scenes/infrared_scene_universal_projector.c]: Use the same title pattern ("X Remote") as in other remotes. Shorten "Projector" if necessary.

**Review by gsurkov — CHANGES_REQUESTED** [review]
Please update [the documentation](https://github.com/flipperdevices/flipperzero-firmware/blob/dev/documentation/UniversalRemotes.md) on adding IR signals to include projectors. Use the existing document as an example.

**Comment by gsurkov** [comment]
@itsweekly we are waiting for the updated library file and the above requested changes, then it will be merged. Un-draft when ready.

**Comment by itsweekly** [comment]
Ready. I have updated the library file, added some of my own collected projectors, and fixed the names and everything you pointed out.

**Review by gsurkov — APPROVED** [review]

**Review by itsweekly — COMMENTED** [review]
Great

**Review by skotopes — APPROVED** [review]
